test/kube_checks.go

```diff
@@ -121,6 +123,31 @@ func countEndpointsNum(e *corev1.Endpoints) int {
 	return num
 }
 
+// GetEndpoints returns addresses of endpoints selected by given listOptions.
+func GetEndpoints(client *KubeClient, namespace string, listOptions metav1.ListOptions) ([]string, error) {
```

Comment:
Is this worth a helper? Should the helper be more composable maybe, i.e.

```golang
func GetEndpointAddresses(eps *corev1.Endpoints) ([]string, error)
```

That avoids threading through the list options which I'm not sure we do anywhere.